sdk/agent/events.go

```diff
@@ -0,0 +1,52 @@
+package agent
+
+import "github.com/stellar/experimental-payment-channels/sdk/state"
+
+type Event interface {
+	event()
+}
+
+// ErrorEvent occurs when an error has occurred, and contains the error
+// occurred.
+type ErrorEvent struct {
+	Err error
+}
+
+func (e ErrorEvent) event() {}
+
+// ConnectedEvent occurs when the agent is connected to another participant.
+type ConnectedEvent struct{}
+
+func (e ConnectedEvent) event() {}
+
+// OpenedEvent occurs when the channel has been opened.
+type OpenedEvent struct{}
+
+func (e OpenedEvent) event() {}
+
+// PaymentReceivedAndConfirmedEvent occurs when a payment is received and has
+// been confirmed as valid.
+type PaymentReceivedAndConfirmedEvent struct {
+	CloseAgreement state.CloseAgreement
+}
+
+func (e PaymentReceivedAndConfirmedEvent) event() {}
+
+// PaymentSentAndConfirmedEvent occurs when a payment is sent and the other
+// participant has confirmed the payment.
```

Comment:
💭  i"m not necessarily tied to this suggestion. But when I first read the comment and variable name I thought the event might be triggered when the other party confirmed not when we confirmed the confirm. Feel free to ignore if it makes it more confusing
```suggestion
// PaymentSentAndConfirmedEvent occurs when a payment is sent, the other
// participant confirmed, and we confirmed.
```

Reply:
> when the other party confirmed not when we confirmed the confirm

I'm not sure the distinction is super important from the caller of the function as that detail of how the confirmations occur exactly is abstracted by the agent. In fact, that would suggest we could rename these events as `PaymentReceived` and `PaymentSent`.

Wdyt?

Reply:
4aae901